src/date/index.ts

```diff
@@ -27,3 +31,344 @@ export function getMonthAbbrByIndex(monthNumber: number): string {
+ * getDateFromLongValue('2022-01-25T12:08:12', 'DD MMM YYYY,  hh:mm A);  // Output will be '25 Jan 2022,  12:08 PM'
+ * getDateFromLongValue('2021/11/20', 'YYYY-MM-DD');  // Output will be '2021-11-20'
+ * getDateFromLongValue('21 Nov 2020', 'YYYY-MM-DD');  // Output will be '2020-11-21'
+ * getDateFromLongValue('Fri Feb 04 2022 15:24:28 GMT+0530 (India Standard Time)', 'YYYY-MM-DD');  // Output will be '2022-02-04'
+ * ```
+ */
+export function getDateFromLongValue(longDateValue: Date, dateFormat: string = 'DD MMM YYYY') {
+  try {
+    if (!isEmpty(dateFormat)) {
+      const str = dayjs(longDateValue).format(dateFormat);
+
+      if (str === 'Invalid Date') {
+        return '';
+
+      } else {
+        return str;
+      }
+    }
+
+  } catch (error) {
+    console.error('Error in getDateFromLongValue: ', error);
+  }
+}
+
+
+/**
+ * This function returns age as a number from the date of birth as an input
+ *
+ * @param {Date} birthDate - birth date of a person in Date format
+ *
+ * @example
+ * ```
+ * getAgeFromDateOfBirth(new Date());  // Output is 0
+ * getAgeFromDateOfBirth('Sun Oct 10 1993 05:30:00 GMT+0530 (India Standard Time)');  // Output is 28
+ * getAgeFromDateOfBirth('1993-10-10');  // Output is 28
+ * getAgeFromDateOfBirth();  // Output is NaN
+ * ```
+ */
+export function getAgeFromDateOfBirth(birthDate: Date) {
+  try {
+    const today = new Date();
+
+    birthDate = new Date(birthDate);
+
+    let age = today.getFullYear() - birthDate.getFullYear();
+    const monthDifference = today.getMonth() - birthDate.getMonth();
+
+    if (monthDifference < 0 || (monthDifference === 0 && today.getDate() < birthDate.getDate())) {
+      age--;
+    }
+
+    return age;
+
+  } catch (error) {
+    console.error('Error im getAgeFromDateOfBirth: ', error);
+  }
+}
+
+
+/**
+ * This method is used for getting previous month with same date to the date provided through argument.
+ * Not doing with dayjs as no specific methods for this use case are there.
+ *
+ * @param {Date} date - Date in valid Date format
+ * @param {number} monthCount - Number of months we want to go back to
+ *
+ * @example
+ * ```
+ * getPreviousMonthDate('Mon Mar 15 2021 10:21:35 GMT+0530 (India Standard Time)');  // Output will be 'Mon Feb 15 2021 10:21:35 GMT+0530 (India Standard Time)'
+ * getPreviousMonthDate('1992-10-10');  // Output will be 'Thu Sep 10 1992 05:30:00 GMT+0530 (India Standard Time)'
+ * getPreviousMonthDate(new Date());  // Output will be 'Tue Jan 04 2022 02:28:43 GMT+0530 (India Standard Time)'
+ * getPreviousMonthDate('03/25/2015');  // Output will be 'Wed Feb 25 2015 00:00:00 GMT+0530 (India Standard Time)'
+ * getPreviousMonthDate();  // Output will be 'Invalid date'
+ * getPreviousMonthDate('03/31/2022');  // Output will be 'Thu Mar 03 2022 00:00:00 GMT+0530 (India Standard Time)'
+ * getPreviousMonthDate(new Date(), 10); // Output will be 'Sun Apr 04 2021 15:18:48 GMT+0530 (India Standard Time)'
```

Comment:
new Date() kya hai .... later we wont understand this. That's why tell new Date is 4th feb and that is why output is 4th april

Reply:
updated